<?xml version="1.0"?>

<launch>
    <!-- File management -->
	<arg name="model_fp" default="../models/yamaha/icra_resnet.pt"/>
    <arg name="obstacle_threshold" default="20.0"/>
    <arg name="publish_gridmap" default="true"/>
    <arg name="device" default="cuda"/>

	<node pkg="maxent_irl_maps" type="gridmap_to_cvar_costmap_node.py" name="maxent_irl_node" output="screen"> 
        <param name="model_fp" value="$(dirname)/$(arg model_fp)"/>
	    <param name="gridmap_topic" value="/local_gridmap"/>
        <param name="costmap_topic" value="/local_costmap"/>
        <param name="odom_topic" value="/integrated_to_init"/>
        <param name="cvar_topic" value="/irl_cvar"/>
        <param name="obstacle_threshold" value="$(arg obstacle_threshold)"/>
        <param name="publish_gridmap" value="$(arg publish_gridmap)"/>
        <param name="device" value="$(arg device)"/>
	</node>

</launch>
